### PR TITLE
Increase version gradio-client version/changelog

### PR DESCRIPTION
**Pull request by WilliamRoyNelson**
# Description

https://github.com/gradio-app/gradio/pull/4230 was accompanied by a push to PyPI as gradio-client==0.2.5
https://pypi.org/project/gradio-client/0.2.5/

This PR did include an update to the changelog, but did not update the version number so the changes were placed under Upcoming Release instead of under 0.2.5

Later, changes were added to the changelog in https://github.com/gradio-app/gradio/pull/4435 but since the version hadn't been incremented, they were marked as upcoming features (presumably for 0.2.5) but won't actually appear until 0.2.6 unless 0.2.5 is overwritten in place.

I have also incremented the gradio requirements file to point to the newest version and incorporate bugfixes.

# Checklist:

- [X] I have performed a self-review of my own code
- [X] I have added a short summary of my change to the CHANGELOG.md
- [X] My code follows the style guidelines of this project
- [X] I have commented my code in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by WilliamRoyNelson** [comment]
@abidlabs @freddyaboulton 
This PR is related to your changes, https://github.com/gradio-app/gradio/pull/4230 and https://github.com/gradio-app/gradio/pull/4435

Can you please ensure that I've gotten this correct? You probably understand your own changes much better than I do.

File: requirements.txt [hunk 1]
```diff
@@ -3,7 +3,7 @@ aiohttp
 altair>=4.2.0
 fastapi
 ffmpy
-gradio_client>=0.2.4
```
Comment (freddyaboulton) [inline requirements.txt:6]: Thanks for the fix @WilliamRoyNelson ! 

This looks good to me but I don't think we actually don't have to bump the version in requirements.txt. gradio will still work with version 0.2.4.

**Comment by abidlabs** [comment]
Thanks @WilliamRoyNelson. As I'm about to release a new version of the Client and the library, I'll merge in this PR minus the requirements change (for now -- we'll update that soon anyways).

**Comment by WilliamRoyNelson** [comment]
> Thanks @WilliamRoyNelson. As I'm about to release a new version of the Client and the library, I'll merge in this PR minus the requirements change (for now -- we'll update that soon anyways).

Thanks for taking care of that for me!
The PyPI pages for both gradio and gradio-client look wonderful now. I appreciate all your help.

**Pull request merged**